Ms 1.3 updates michael (#76)
BitcoinDesign/Bitcoin-Core-App: pages/milestones/1-3-import.md

Below is an iteration of the wallet import flow, including migration and error paths. Many details need to be refined. Click to see it larger.

{% include picture.html

> 	image = "/assets/images/import-wallet/flow.png"

Reviewer: If you remove a reference to an image and the image is not used anymore, please remove the image file itself from the repo as well.

> 	image = "/assets/images/import-wallet/import_flow.png"

Reviewer: Regular image should be 800px wide. Retina twice that.

> 	retina = "/assets/images/import-wallet/flow@2x.png"

Reviewer: Make sure to also replace the retina image.

> 	big = "/assets/images/import-wallet/import_flow.png"

Reviewer: Big size cannot be the same as the regular size. It should be high-res enough that someone can dig in and see all the details. For flows, that can end up being quite big.

	alt-text = "User flow mock-ups for importing an existing wallet."
	width = 800

> 	height = 368

Reviewer: Make sure to update the height once you changed the images to the right dimensions.

%}

## Entry points

For those users who would like to use Bitcoin Core with a wallet, the app offers three main entry points for adding wallets:

> - Directly after the onboarding flow, if the user is doing a fresh install of the application.

Reviewer: ```suggestion
- Directly at the end of the onboarding flow, if the user is doing a fresh install of the application.
```

> - Through the wallet selector dropdown, if he already has a different wallet. 

Reviewer: ```suggestion
- Through the wallet selector dropdown, if a wallet was already previously added. 
```

> - From the settings screen, if the application has been previously used in node only mode. 

Reviewer: Didn't we just have a small add wallet button in the top-left corner?

{% include picture.html
	image = "/assets/images/import-wallet/entry-points.png"

> 	retina = "/assets/images/import-wallet/entry-points@2x.png"

Reviewer: big image missing

	alt-text = "Mockups showing the entry points to the wallet import flow."
	width = 800

> 	height = 368

Reviewer: Make sure to update the height.

%}

## Import options

> Users have two main options to import a wallet: by loading a wallet file into the application or by copy-pasting an extended public key, bitcoin address or descriptor.

Reviewer: ```suggestion
Users have two main options to import a wallet: by loading a wallet file into the application or by entering an extended public key, bitcoin address or descriptor.
```

{% include picture.html
	image = "/assets/images/import-wallet/import-wallet-options.png"
	retina = "/assets/images/import-wallet/import-wallet-options@2x.png"

> 	alt-text = "Mockup showing the import screen."

Reviewer: big image missing and wrong image height.

	width = 400
	height = 184
%}

#### File upload

> Users can import a wallet by choosing a wallet file from their file system. In that case, the native file picker from their operating system. On desktop they can also just drag and drop the file into the application window. 

Reviewer: ```suggestion
Users can import a wallet by choosing a wallet file from their file system (via the native operating system file picker). On desktop, they can also drag and drop the file into the application window. 
```

> #### Paste XPUB, bitcoin address or wallet descriptor

Reviewer: ```suggestion
#### Copy & paste
```

> Users can also copy and paste an XPUB, bitocin address or wallet descriptor to import their wallet.

Reviewer: ```suggestion
Users can also copy and paste an extended public key (XPUB), bitcoin address or wallet descriptor to import their wallet.
```

## Wallet review screen

After a wallet has been imported, the application displays the most important information and metadata about the wallet, so that the user can quickly assess whether everything worked as expected. The information displayed includes: 

- Wallet name (if available)
- Key scheme
- Address type

> - Potentially more

Reviewer: ```suggestion
- Potentially more  based on wallet type
```

> {% include picture.html

Reviewer: big image missing and image height incorrect.

	image = "/assets/images/import-wallet/import-review-screen.png"
	retina = "/assets/images/import-wallet/import-review-screen@2x.png"
	alt-text = "Mockup showing the wallet review screen for a single-key and a multi-key wallet."
	width = 400
	height = 184
%}

> The full details are available in the settings section for full review. Note that the information displayed this screen will vary based on the specific use case. For intance, if a user imports from an XPUB, the application will not now the wallet name.

Reviewer: ```suggestion
The full details are available in the settings section for full review. The information displayed on this screen will vary based on the specific use case. For instance, if a user imports from an XPUB, the application will not know the wallet name.
```

## Exception and error handling

So far we discussed the happy path. However, there are various alternative scenarios to consider. These include:

- The user tries to import a wallet file that has an outdated format.
- The provided information (wallet file or pasted string) could not be processed.

> - Etc.

Reviewer: ```suggestion
```

{% include picture.html
	image = "/assets/images/import-wallet/error-handling.png"
	retina = "/assets/images/import-wallet/error-handling@2x.png"
	alt-text = "Mockup showing a warning and an error message."
	width = 400

> 	height = 184

Reviewer: big image missing and image size needs updated

%}

In all of these cases, we want to make sure that the application provides clear, concise and actionable feedback about:
1. What went wrong (reason).
2. What this means (impact).

> 3. What does the user need to do (remedy).

Reviewer: ```suggestion
3. What the user can do (remedy).
```
